internal/graph/singleflight_resolver.go

```diff
@@ -58,11 +65,13 @@ func (s *singleflightCheckResolver) ResolveCheck(
 		isUnique = true
 		resp, err := s.delegate.ResolveCheck(ctx, req)
 		if err != nil {
+			telemetry.TraceError(span, err)
```

Comment:
nit: this one is redundant, it will be caught by 74 :)

Reply:
Yes, good catch. Also noticed that the resolver state should be moved up before the error check so that it can properly set that info on the trace if the error is non-nil.